### DISCUSSION TITLE
Add new sensor device class: Wind

### DISCUSSION THREAD

**Discussion by riokuu** [home-assistant/architecture#775] [Entity Models] [answered]
Hi, I am working for a company whose new product is a Wind sensor. I want to add new integration for this device classes, as some users request. However, there is no reasonable class that could match this. This class doesn't fit to Weather entity as it's a single sensor. From interesting applications of this sensor, one user used it in an aerodynamic tunnel. There was one application in jet engine trust tests.

For the _Wind_ device class, I propose:

- To extend  `SensorDeviceClass(StrEnum)` by:

    `WIND = 'wind'`
(unit of measurement can use Speed units from _**const.py**_ )
- and to complement, add to `Speed units` in _**homeassistant/const.py**_:

    `SPEED_KNOTS: Final = "kn"`
    
   It would be beneficial for any integrations that would like to implement Wind sensors in the future.

**Comment by MartinHjelmare** [comment]
Please do some market research of the current integrations and prominent available ones in the market and give an estimate how many integrations can benefit from this new device class. Please make a list with names. Thanks!

We always ask this question before adding new device classes. See this discussion for a reference:
https://github.com/home-assistant/architecture/discussions/688#discussioncomment-1805643

**Reply by farmio** [reply]
KNX could make use of it for Datapoint types 
- DPT 9.005 Wind Speed in m/s
- DPT 9.028 Wind Speed km/h
- DPT 20.014 Beaufort Wind force scale

**Reply by riokuu** [reply]
- Blebox multiSensor with wind sensor,
- AcuRite,
- Windy API, 
- Netatmo 
Above a variety of potential applications(I've checked if documentation provides API for wind speed data), I believe that further research would bring more to the table.

**Reply by MartinHjelmare** [reply]
Looks like potential use is enough for adding this device class.

@emontnemery what do you think?

**Reply by iMicknl** [reply]
Overkiz also has a wind sensor; however I am not sure of the `native_unit_of_measurement`... I would need to find a user with this sensor.

https://github.com/home-assistant/core/blob/9a0b3796d3fcf45b384604f10cb4367f27b0facb/homeassistant/components/overkiz/sensor.py#L302-L309

**Reply by emontnemery** [reply]
I think this is a good proposal, but I think the new device class should be named `wind_speed`.

**Comment by Jc2k** [comment]
There was some conversation on another thread (and on discord) about what we would need to do to break out some of the weather data as separate entities. Having a device class for wind could be useful for that effort too, although it's still just an idea atm.

CC @emontnemery

**Comment by emontnemery** [comment]
Updated proposal; `wind` renamed to `wind_speed` and knot unit removed because it's already implemented:

- Extend  `SensorDeviceClass(StrEnum)` by:

    `WIND_SPEED = 'wind_speed'`
(unit of measurement can use `SPEED_*` units from _**const.py**_ )

- Implement automatic unit conversion according to the unit system; km/h for metric and mph for imperial.

- Implement custom unit selection option for the user and unit conversion to go with it, similar to what we've done for the temperature sensor device class.

It's ok to go ahead with PRs that do this, for core, frontend, dev docs and user docs.

**Reply by epenet** [reply]
I think it makes sense to have this as a separate device class, with a specific icon.
The unit people use for wind speed may be different form the unit used for other types of speed (car, etc.)

**Reply by epenet** [reply]
PR opened: https://github.com/home-assistant/core/pull/79789

**Reply by frenck** [reply]
This proposal makes no sense to me. The only reason for adding this device class at this point: Is a default icon? I don't think that is justification for a backend change.

**Reply by epenet** [reply]
These are reasons that I think makes this relevent:
- the first in my opinion is "logical grouping" (there are lots of wind speed sensors inside home assistant)
- adds a default icon (as you mentionned)
- restricts the units of measurement (m/s, km/h, mph) and doesn't show other units

A separate device class will also make it easier to choose a default display unit of measurement if (/when?) we enable automatic conversion

_side note: I always felt that new device classes were added based on how many components would make use of it. I think `frequency`, `current` and `voltage` in effect only add a default icon._

**Reply by frenck** [reply]
> the first in my opinion is "logical grouping" 

The is no such concept in Home Assistant.

> adds a default icon (as you mentioned)

Default icons can be overridden / provided better ones.

> restricts the units of measurement (m/s, km/h, mph) and doesn't show other units

![image](https://user-images.githubusercontent.com/195327/194808605-0d735174-f38f-4436-93ed-5fdcc9e91401.png)

These are also supported by `speed`, there is no real reason to create the limitation you are suggesting.

**Comment by emontnemery** [comment]
Alternative proposal:

There's no need for a specific sensor device class for wind speed; it's enough with a broader sensor device class for speeds in general as proposed in https://github.com/home-assistant/architecture/discussions/798

**Reply by frenck** [reply]
Sounds more sane to me.

**Comment by frenck** [comment, marked as answer]
`wind_speed` has been added in https://github.com/home-assistant/core/pull/79789

As it provides a base to provide default units in the future.
